icicle/fields/bn254/extern.cu

```diff
@@ -0,0 +1,8 @@
+#define FIELD bn254
```

Comment:
ideally we should compile those three files in parallel and not include source files. Only headers (cuh) should be included.
I know it's tricky for template libraries but we can explicitly instantiate where need to.
The way you chose is slower and I think should be avoided if possible.

Reply:
you are right. I'll change it